### PR TITLE
Refactor of links in Dataproc

### PR DESCRIPTION
**Pull request by bkossakowska**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/google/cloud/links/dataproc.py [hunk 1]
```diff
@@ -20,99 +20,147 @@
...
+    key = "dataproc_cluster"
+    format_str = DATAPROC_CLUSTER_LINK
 
     @staticmethod
     def persist(
         context: Context,
-        task_instance,
-        url: str,
-        resource: str,
+        task_instance: BaseOperator,
```
Comment (uranusjr) [inline airflow/providers/google/cloud/links/dataproc.py]: Is this correct? A variable named `task_instance` should generally be a `TaskInstance` object instead. If the type is accurate, we should consider renaming the argument.

File: airflow/providers/google/cloud/links/dataproc.py [hunk 2]
```diff
@@ -20,99 +20,151 @@
...
+    key = "dataproc_cluster"
+    format_str = DATAPROC_CLUSTER_LINK
 
     @staticmethod
     def persist(
         context: Context,
-        task_instance,
-        url: str,
-        resource: str,
+        operator_instance: BaseOperator,
```
Comment (uranusjr) [inline airflow/providers/google/cloud/links/dataproc.py]: Please call this `task` or `operator` instead; the object is too obviously an instance and the name shouldn’t be lengthened unnecessarily.

**Comment by potiuk** [comment]
conflicts :(

**Comment by eladkal** [comment]
@bkossakowska can you have the conflicts and rebase the PR?

**Comment by bkossakowska** [comment]
> @bkossakowska can you have the conflicts and rebase the PR?

Hi @eladkal,
PR is rebased

**Comment by VladaZakharova** [comment]
Hi @potiuk !
Can we review changes? Thanks!

**Comment by VladaZakharova** [comment]
Hi @uranusjr ! Could you please recheck this PR? Thanks!

**Comment by VladaZakharova** [comment]
Hi @potiuk  @eladkal !
Could you please check the changes? :)

**Comment by bkossakowska** [comment]
Hi @potiuk,
could you please take a look at the changes? Thanks :)

**Review by potiuk — APPROVED** [review]

**Pull request merged**